File: prototype/rails/src/app/domain/listings/listing_event_type.rb
module Domain
  module Listings
    module ListingEventType
      VIEW = "view"
      FAVORITE = "favorite"
      UNFAVORITE = "unfavorite"
      CART_ADD = "cart_add"

      ALL = [VIEW, FAVORITE, UNFAVORITE, CART_ADD].freeze
    end
  end
end

File: prototype/php/src/app/Domain/Listings/ListingEventType.php
<?php

declare(strict_types=1);

namespace App\Domain\Listings;

enum ListingEventType: string
{
    case View = 'view';
    case Favorite = 'favorite';
    case Unfavorite = 'unfavorite';
    case CartAdd = 'cart_add';

    public function label(): string
    {
        return ucfirst(str_replace('_', ' ', $this->value));
    }
}
